src/Components/AddComment.jsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import { postCommentsForArticle } from "../utils/api";
+
+export default function AddComment({ setComments, article_id }) {
+  const [newComment, setNewComment] = useState({ username: "", body: "" });
+  const [isError, setIsError] = useState("");
+
+  const handleOnSubmit = (e) => {
+    e.preventDefault();
+    setComments((currentComments) => {
+      const optimisticComment = { ...newComment, comment_id: Date.now() };
+      return [optimisticComment, ...currentComments];
+    });
+    postCommentsForArticle(article_id, newComment)
+      .then((data) => {})
+      .catch((err) => {
+        setIsError(err.response.data.msg);
```

Comment:
I probably wouldnt do it like this, and add the comment that comes back from the database. 

At the moment, i can add a comment from a user that isnt there, and it appears on the page. 
it doesnt persist or get added to the db, but from a UX point of view, it looks like it has been successful, until you refresh. 
if you use the database data, this wont happen, as it wont be successful. 
You can even have a little posting... icon while the request is happening